### ISSUE TITLE
IDN-domains not displayed correctly

### ISSUE THREAD

**Issue by kevinkk525** [nextcloud/android#1046]
### Actual behaviour
-IDN-domains (https://www.denic.de/en/know-how/idn-domains/) do work but in the account name they are displayed in their ACE string instead of IDN version.
Of course this is no bug, just a visual improvement.

### Expected behaviour
-Display IDN-domains correctly
 
### Steps to reproduce
1. Get an IDN-domain
2. Log in
3. check account name

### Environment data
Android version: 7.1.2

Device model:  Oppo Find 7

Nextcloud app version: Nextcloud-dev 20170525

Nextcloud server version: 12.0.0

**Comment by AndyScherzinger** [comment]
Hi @kevinkk525, thanks for the input! And yes, while it is technically not a but I agree that this doesn't look very nice and see below, I also think this be fixed (even though it might be helpful if you could provide us with a test account so we can test the code change once it is done, waiting for the other folks approval first though).

@mario @tobiasKaminsky sounds good to me, a quick search revealed that this might be achievable using 
http://download.java.net/jdk7/archive/b123/docs/api/java/net/IDN.html#toUnicode(java.lang.String) - not sure if the setting of the flags might be necessary.

**Comment by AndyScherzinger** [comment]
@kevinkk525 Or if you are willing to test we can first deploy it to the dev builds :) ...which might be easier!

**Comment by kevinkk525** [comment]
You can deploy it to dev builds as I'm using those already anyways. I will gladly test it.
Additionally I can still create a test account for you if needed. 
Am 28.05.17, 00:30, Andy Scherzinger <notifications@github.com> schrieb:
@kevinkk525 Or if you are willing to test we can first deploy it to the dev builds :) ...which might be easier!

**Comment by kevinkk525** [comment]
Just bumping this up as no progress was made during the last 2 years. (I know it's not a bug and has low priority).

**Comment by AndyScherzinger** [comment]
Hi @kevinkk525, thanks for the reminder 👍
Would you be willing to create a test account for us and send it to either @tobiasKaminsky or me via email? 
I'd then work on a solution for it 😊

**Comment by kevinkk525** [comment]
Thanks for working on this. I created a testaccount and sent the login to Tobias as I couldn't find a way to contact you personally.

**Comment by AndyScherzinger** [comment]
Thanks! Today is a bank holiday in Germany, so @tobiasKaminsky will likely forward it to me by tomorrow :)

**Comment by tobiasKaminsky** [comment]
Forwarded and assigned Andy

**Comment by AndyScherzinger** [comment]
@kevinkk525 @tobiasKaminsky I created a fix, see #3948 👍 

@kevinkk525 please keep the account active so @tobiasKaminsky can test my implementation ❤️ Thanks for providing a test account. 🙏

**Comment by tobiasKaminsky** [comment]
Tested. @kevinkk525 you can remove the test account.

**Comment by kevinkk525** [comment]
Thanks a lot for implementing this.

**Comment by AndyScherzinger** [comment]
Fix has also been backported and will ship with 3.6.1 too.
